### PR TITLE
Add bluemicro support

### PR DESCRIPTION
**Pull request by Na-Cly**
Added support for BlueMicro52840_V1

### CODE REVIEW

File: app/boards/arm/bluemicro52840_v1/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Electronut Labs Papyr board configuration
```
Comment (Nicell) [inline app/boards/arm/bluemicro52840_v1/Kconfig.defconfig:1]: This seems out of place... I see it in the nice!nano def too. Oops?

File: app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	tx-pin = <6>;
+	rx-pin = <8>;
+};
+
+&i2c0 {
+	compatible = "nordic,nrf-twi";
+	sda-pin = <25>;
+	scl-pin = <26>;
```
Comment (Nicell) [inline app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts:52]: I think this should be 15 and 17?

**Comment by Nicell** [comment]
#90 should probably be done for this board as well

File: app/boards/arm/bluemicro52840_v1/arduino_pro_micro_pins.dtsi [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
```
Comment (petejohanson) [inline app/boards/arm/bluemicro52840_v1/arduino_pro_micro_pins.dtsi:2]: This should be you!

File: app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	rx-pin = <8>;
+};
+
+&i2c0 {
+	compatible = "nordic,nrf-twi";
+	sda-pin = <15>;
+	scl-pin = <17>;
+};
+
+&uart0 {
```
Comment (petejohanson) [inline app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts:55]: You're overriding `uart0` *twice* in here, this one, and the one above. Which one is the right one? (if you inherited this from the nano board, we likely need to fix that too)

File: app/boards/arm/bluemicro52840_v1/board.cmake [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
```
Comment (petejohanson) [inline app/boards/arm/bluemicro52840_v1/board.cmake:1]: This license should be fixed, `MIT`.

File: docs/static/setup.sh [hunk 6]
```diff
@@ -11,7 +11,7 @@ title="ZMK Config Setup:"
 # TODO: Check for user.name and user.email git configs being set
 
 prompt="Pick an MCU board:"
-options=("nice!nano" "QMK Proton-C")
+options=("nice!nano" "QMK Proton-C" "bluemicro52840_v1")
```
Comment (petejohanson) [inline docs/static/setup.sh:14]: This should be the "friendly name", e.g.:

```suggestion
options=("nice!nano" "QMK Proton-C" "BlueMicro52840 (v1)")
```

Also, you need to add a selection in the `select` block below that will set `board="bluemicro52840_v1"`.

File: docs/static/setup.sh [hunk 7]
```diff
@@ -37,7 +37,7 @@ echo ""
 echo "Keyboard Shield Selection:"
 
 prompt="Pick an keyboard:"
-options=("Kyria" "Lily58" "Corne")
+options=("Kyria" "Lily58" "Corne" "splitreus62")
```
Comment (petejohanson) [inline docs/static/setup.sh:40]: Please add this in the other PR for this shield, thanks!

Comment (Na-Cly) [reply docs/static/setup.sh:40]: Will do.

File: app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts (hunk 4 above)
Comment (Na-Cly) [reply app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts:55]: Yeah it's directly from the nice_nano.

File: app/boards/arm/bluemicro52840_v1/Kconfig.defconfig (hunk 1 above)
Comment (petejohanson) [reply app/boards/arm/bluemicro52840_v1/Kconfig.defconfig:1]: Yeah, this should be fixed, and we should fix in the nano board too.

File: app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts (hunk 4 above)
Comment (petejohanson) [reply app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts:55]: Ok, then removed the first `uart0` node that is above this one please.

File: docs/static/setup.sh [hunk 8]
```diff
@@ -11,7 +11,7 @@ title="ZMK Config Setup:"
 # TODO: Check for user.name and user.email git configs being set
 
 prompt="Pick an MCU board:"
-options=("nice!nano" "QMK Proton-C")
+options=("nice!nano" "QMK Proton-C" "BlueMicro52840 (v1)")
```
Comment (petejohanson) [inline docs/static/setup.sh:14]: Still missing the extra section in the `select` block right below this.

Comment (Na-Cly) [reply docs/static/setup.sh:14]: Added.

File: app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts (hunk 4 above)
Comment (petejohanson) [reply app/boards/arm/bluemicro52840_v1/bluemicro52840_v1.dts:48]: Removed.

**Pull request merged**